vme/src/event.cpp

```diff
@@ -287,8 +285,8 @@ void eventqueue::process()
                     {
                         strcpy(dilzname, prg->fp->tmpl->zone->getName());
                     }
-                    strcpy(diloname, tmp_event->arg1 ? UNIT_FI_NAME((unit_data *)(tmp_event->arg1)) : "NO NAME");
-                    strcpy(dilozname, tmp_event->arg1 ? UNIT_FI_ZONENAME((unit_data *)(tmp_event->arg1)) : "NO ZONE");
+                    strcpy(diloname, u ? u->getFileIndexName() : "NO NAME");
```

Comment:
A possible enhancement would be that getFIleIndexName() returns "NO NAME", and ZoneName() returns "NO ZONE" in case the fileIndex is null. That would clean up a few places like this.